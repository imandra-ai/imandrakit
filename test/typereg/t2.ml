open Util_

[@@@ocaml.warning "-37-69"]

type foo = {
  x: int;
  y: float option;
}
[@@deriving typereg]

type bar =
  | A
  | B of int
  | C of int * string
  | D of {
      x: (int[@twine_skip_field]);
      y: (string * foo) list;
    }
[@@deriving typereg]

let () =
  print_endline "t2:";
  dump ()
