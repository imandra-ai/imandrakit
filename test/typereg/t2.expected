t2:
{ path = "Dune__exe__T2"; name = "bar"; params = [];
  decl =
  (Alg
     [{ c = "A"; args = []; labels = None };
       { c = "B"; args = [(Cstor ("int", []))]; labels = None };
       { c = "C"; args = [(Cstor ("int", [])); (Cstor ("string", []))];
         labels = None };
       { c = "D";
         args =
         [(Cstor ("int", []));
           (Cstor ("list",
              [(Tuple [(Cstor ("string", [])); (Cstor ("foo", []))])]))
           ];
         labels = (Some ["x"; "y"]) }
       ])
  }
{ path = "Dune__exe__T2"; name = "foo"; params = [];
  decl =
  (Record
     { fields =
       [("x", (Cstor ("int", [])));
         ("y", (Cstor ("option", [(Cstor ("float", []))])))]
       })
  }
